packages/nitro-protocol/src/contract/challenge.ts

```diff
@@ -1,9 +1,57 @@
 import {Bytes32} from './types';
-import {keccak256, defaultAbiCoder} from 'ethers/utils';
-import {State, hashState} from './state';
+import {keccak256, defaultAbiCoder, bigNumberify} from 'ethers/utils';
+import {hashState, VariablePart, State} from './state';
+import {Channel} from '..';
+import {decodeOutcome} from './outcome';
 
 export function hashChallengeMessage(challengeState: State): Bytes32 {
   return keccak256(
     defaultAbiCoder.encode(['bytes32', 'string'], [hashState(challengeState), 'forceMove']),
   );
 }
+
+export interface ChallengeRegisteredEvent {
+  challengerAddress: string;
+  finalizesAt: string;
+  challengeStates: State[];
+}
+export function getChallengeRegisteredEvent(eventResult): ChallengeRegisteredEvent {
+  const [event] = eventResult.slice(-1);
+  const {
+    turnNumRecord,
+    finalizesAt,
+    challenger,
+    isFinal,
+    fixedPart,
+    variableParts: variablePartsUnstructured,
+  } = event.args;
+
+  // Fixed part
+  const chainId = bigNumberify(fixedPart[0]).toHexString();
+  const participants = fixedPart[1].map(p => bigNumberify(p).toHexString());
+  const channelNonce = bigNumberify(fixedPart[2]).toHexString();
+  const appDefinition = fixedPart[3];
+  const challengeDuration = bigNumberify(fixedPart[4]).toNumber();
```

Comment:
This decoding will certainly be getting re-used in the handlers for other events. I would recommend stripping it out into a short re-usable function.